### PR TITLE
DOP-4713 unit test

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

DOP-4713

### Notes:
- No behavioral changes
- Pre-req PR is here that actually implements the dropdown
- Tests to validate the rendered button, rendered menu, behavior of switching to dark mode, and consumption of dark mode

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [X] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

File: tests/unit/DarkModeDropdown.test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  it('renders dark mode dropdown', async () => {
+    // first snapshot of closed menu
+    const elm = mountDarkModeDropdown();
+    expect(elm.asFragment()).toMatchSnapshot();
+    const button = await elm.findByLabelText('Dark Mode Menu');
+    await act(async () => {
+      button.click();
+    });
+    // second snapshot of open menu
+    expect(elm.asFragment()).toMatchSnapshot();
```
Comment (rayangler) [inline tests/unit/DarkModeDropdown.test.js:43]: Might be a nit, but can we leave a comment on what the expected difference is between the two snapshots? If the test changes in any way and we need to make sure the outputs are adjusted accordingly, the comment could save us time from needing to go through the entire snapshots

Comment (rayangler) [reply tests/unit/DarkModeDropdown.test.js:43]: Nevermind, I can't read. I realized what the menus were now.

Comment (seungpark) [reply tests/unit/DarkModeDropdown.test.js:43]: glad the comments helped! 😛

File: tests/unit/DarkModeDropdown.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+// mock window.localStorage
+const storage = {};
+jest.spyOn(window.localStorage.__proto__, 'setItem').mockImplementation((key, value) => {
+  storage[key] = value;
+});
+
+const mountDarkModeDropdown = () => {
+  return render(
+    <DarkModeContext.Provider value={{ setDarkModePref, darkModePref }}>
+      <LeafyGreenProvider baseFontSize={16} darkMode={darkModePref === 'dark-theme' ? true : false}>
```
[the remark this answers is not in the document]
Comment (seungpark) [reply tests/unit/DarkModeDropdown.test.js:26]: good catch! a bit different from the context but this is testing for the dropdown

**Review by rayangler — APPROVED** [review]

**Pull request merged**